cmd/csi-addons/replication.go

```diff
@@ -0,0 +1,245 @@
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.DisableVolumeReplicationRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.DisableVolumeReplication(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Disable Volume Replication operation successful")
+
+	return nil
+}
+
+// PromoteVolume executes the PromoteVolume operation.
+type PromoteVolume struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("PromoteVolume", &PromoteVolume{})
+
+func (rep *PromoteVolume) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.PromoteVolumeRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.PromoteVolume(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Promote Volume operation successful")
+
+	return nil
+}
+
+// DemoteVolume executes the DemoteVolume operation.
+type DemoteVolume struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("DemoteVolume", &DemoteVolume{})
+
+func (rep *DemoteVolume) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.DemoteVolumeRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.DemoteVolume(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Demote Volume operation successful")
+
+	return nil
+}
+
+// ResyncVolume executes the ResyncVolume operation.
+type ResyncVolume struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("ResyncVolume", &ResyncVolume{})
+
+func (rep *ResyncVolume) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.ResyncVolumeRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.ResyncVolume(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Resync Volume operation successful")
+
+	return nil
+}
+
+// GetVolumeReplicationInfo executes the GetVolumeReplicationInfo operation.
+type GetVolumeReplicationInfo struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("GetVolumeReplicationInfo", &GetVolumeReplicationInfo{})
+
+func (rep *GetVolumeReplicationInfo) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.GetVolumeReplicationInfoRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.GetVolumeReplicationInfo(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("GetVolumeReplicationInfo operation successful")
```

Comment:
Get should print the response that you got from the API as its meant to display details. i would suggest sending a PR to fix it. for now approving this

Reply:
Sure, I'll send a PR for same.
Thank you!